aptos-move/framework/move-stdlib/src/natives/bcs.rs

```diff
@@ -74,13 +76,62 @@ fn native_to_bytes(
     Ok(smallvec![Value::vector_u8(serialized_value)])
 }
 
+/***************************************************************************************************
+ * native fun serialized_size
+ *
+ *   gas cost: size_of(output)
+ *
+ *   If the getting the type layout or serialization results in error, a special failure
+ *   cost is charged.
+ *
+ **************************************************************************************************/
+fn native_serialized_size(
+    context: &mut SafeNativeContext,
+    mut ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    debug_assert!(ty_args.len() == 1);
+    debug_assert!(args.len() == 1);
+
+    let reference = safely_pop_arg!(args, Reference);
+    let ty = ty_args.pop().unwrap();
+
+    let serialized_size = match serialized_size_impl(context, reference, &ty) {
+        Ok(serialized_size) => serialized_size as u64,
+        Err(_) => {
+            context.charge(BCS_SERIALIZED_SIZE_FAILURE)?;
+
+            // Re-use the same abort code as bcs::to_bytes.
+            return Err(SafeNativeError::Abort {
+                abort_code: NFE_BCS_SERIALIZATION_FAILURE,
+            });
+        },
+    };
+    context.charge(BCS_SERIALIZED_SIZE_PER_BYTE_SERIALIZED * NumBytes::new(serialized_size))?;
+
+    Ok(smallvec![Value::u64(serialized_size)])
+}
+
+fn serialized_size_impl(
+    context: &mut SafeNativeContext,
+    reference: Reference,
+    ty: &Type,
+) -> PartialVMResult<usize> {
+    let value = reference.read_ref()?;
```

Comment:
`read_ref` performs a deep copy of the value. As the value never leaves the native I wonder if we can implement it in a more efficient way. 

Maybe we can implement a new `fn get_ref()`, which uses `copy_by_ref` internally?

Reply:
I would leave it as a TODO: will create a ticket for this.

Reply:
Sounds good